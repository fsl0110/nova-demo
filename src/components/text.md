### Set text blocks with optional teaser text and selectable topic tag.

#### With topicTag h1

```jsx
<Text
  topicTag="h1"
  topic="Hello text component."
  text="Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua."
/>
```

#### With topicTag h6

```jsx
<Text
  topicTag="h6"
  topic="Hello text component."
  text="Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua."
/>
```

#### With topicTag h1 and teaser

```jsx
<Text
  topicTag="h1"
  topic="Hello text component."
  teaser="Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."
  text="Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua."
/>
```
