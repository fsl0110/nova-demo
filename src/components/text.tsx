import React, { FC } from "react";
import classNames from "classnames";
import dompurify from "dompurify";
import { Tag } from "./tag";
import { SharedProps } from "../typings";

const sanitizer = dompurify.sanitize;

export interface Props extends SharedProps {
  /** HTML-tag used around topic text**/
  topicTag?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
  /** Heading text between the your selected topicTag. Is no topic is set, a topicTag will also not set.**/
  topic?: string;
  /** Teaser text between strong tags **/
  teaser?: string;
  /** Paragraph text between p tags **/
  text?: string;
}

export const Text: FC<Props> = ({
  topicTag = "h1",
  topic,
  teaser,
  text,
  className
}) => (
  <div className={classNames("text", className)}>
    {topic && <Tag type={topicTag}>{topic}</Tag>}
    {teaser && (
      <strong dangerouslySetInnerHTML={{ __html: sanitizer(teaser) }} />
    )}
    {text && <p dangerouslySetInnerHTML={{ __html: sanitizer(text) }} />}
  </div>
);

export default Text;
