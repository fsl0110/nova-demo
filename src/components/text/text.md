### Set text blocks with optional teaser text and selectable topic tag.

#### Text with TopicTag h1

```jsx
<Text
  topicTag="h1"
  topic="Hello text component."
  text="Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua."
/>
```

#### Text with topicTag h6

```jsx
<Text
  topicTag="h6"
  topic="Hello text component."
  text="Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua."
/>
```

#### Text with topicTag h1 and teaser

```jsx
<Text
  topicTag="h1"
  topic="Hello text component."
  teaser="Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."
  text="Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua."
/>
```

#### Text with only topic

```jsx
<Text topicTag="h1" topic="Hello text component." />
```

#### Text with only text

```jsx
<Text text="Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua." />
```

#### Text with only teaser

```jsx
<Text teaser="Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet." />
```
