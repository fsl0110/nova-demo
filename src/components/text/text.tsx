import React, { FC, HTMLAttributes } from "react";
import classNames from "classnames";
import dompurify from "dompurify";
import { Tag } from "..";

const sanitizer = dompurify.sanitize;

export interface Props extends HTMLAttributes<HTMLDivElement> {
  /** HTML-tag used around topic text**/
  topicTag?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
  /** Heading text between the your selected topicTag. Is no topic is set, a topicTag will also not set.**/
  topic?: string;
  /** Teaser text between strong tags **/
  teaser?: string;
  /** Paragraph text between p tags **/
  text?: string;
}

export const Text: FC<Props> = ({
  topicTag = "h1",
  topic,
  teaser,
  text,
  className
}) => (
  <div className={classNames("text", className)} data-test="Text">
    {topic && <Tag type={topicTag}>{topic}</Tag>}
    {teaser && (
      <strong dangerouslySetInnerHTML={{ __html: sanitizer(teaser) }} />
    )}
    {text && <p dangerouslySetInnerHTML={{ __html: sanitizer(text) }} />}
  </div>
);

export default Text;
