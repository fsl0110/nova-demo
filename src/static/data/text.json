{
  "reports": {
    "topic": "Food recall enforcement reports since 2012",
    "teaser": "",
    "text": "This is the openFDA API endpoint for all food product recalls monitored by the FDA. When an FDA-regulated product is either defective or potentially harmful, recalling that product—removing it from the market or correcting the problem—is the most effective means for protecting the public.",
    "ChartTitle": "Enforcement reports over time"
  },
  "initiators": {
    "topic": "The vast majority of food recalls are voluntary",
    "teaser": "",
    "text": "Manufacturers and/or distributors may initiate a recall at any time to fulfill their responsibility to protect the public health from products that present a risk of injury or gross deception, or are otherwise defective. Firms may also initiate a recall following notification of a problem by FDA or a state agency, in response to a formal request by FDA, or as ordered by FDA.",
    "chartTitle": "Enforcement initiators"
  }
}
