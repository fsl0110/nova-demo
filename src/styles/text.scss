.text {
  /*  margin-bottom: 0.5rem; */
}
.text h3,
.text h2,
.text h1 {
  margin-top: 0;
  margin-bottom: 0.8rem;
}

.text p {
  font-size: 1.1rem;
  line-height: 1.4rem;
  margin: 0;
  padding: 0 0 0.5rem 0;
}

.text--two-columns {
  columns: auto 2;
}
.text--three-columns {
  columns: auto 3;
}

@media (max-width: 1200px) {
  .text--two-columns {
    columns: auto 1;
  }
}
