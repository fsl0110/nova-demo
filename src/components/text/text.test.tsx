import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Text, Props } from "./text";

const testCases: Array<[string, Props]> = [
  [
    "Text with h1 topic and teaser",
    {
      topicTag: "h1",
      topic: "Hello Jest",
      teaser: "This is a teaser text.",
      text: "Let's write some tests"
    }
  ],
  [
    "Text with h2 topic",
    {
      topicTag: "h2",
      topic: "Hello Jest",
      text: "Let's write some tests"
    }
  ],
  [
    "Text null",
    {
      topicTag: "h3"
    }
  ],
  [
    "Text with only text",
    {
      text: "Let's write some tests"
    }
  ],
  [
    "Text with only teaser",
    {
      teaser: "This is a teaser text."
    }
  ]
];

describe("<Text>", () => {
  testCases.forEach(([item, options]) => {
    it(item, () => {
      expect(toJson(shallow(<Text {...options} />))).toMatchSnapshot();
    });
  });
});
