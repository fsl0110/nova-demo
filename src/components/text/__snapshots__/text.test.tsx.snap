// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> Text null 1`] = `
<div
  className="text"
  data-test="Text"
/>
`;

exports[`<Text> Text with h1 topic and teaser 1`] = `
<div
  className="text"
  data-test="Text"
>
  <Tag
    type="h1"
  >
    Hello Jest
  </Tag>
  <strong
    dangerouslySetInnerHTML={
      Object {
        "__html": "This is a teaser text.",
      }
    }
  />
  <p
    dangerouslySetInnerHTML={
      Object {
        "__html": "Let's write some tests",
      }
    }
  />
</div>
`;

exports[`<Text> Text with h2 topic 1`] = `
<div
  className="text"
  data-test="Text"
>
  <Tag
    type="h2"
  >
    Hello Jest
  </Tag>
  <p
    dangerouslySetInnerHTML={
      Object {
        "__html": "Let's write some tests",
      }
    }
  />
</div>
`;

exports[`<Text> Text with only teaser 1`] = `
<div
  className="text"
  data-test="Text"
>
  <strong
    dangerouslySetInnerHTML={
      Object {
        "__html": "This is a teaser text.",
      }
    }
  />
</div>
`;

exports[`<Text> Text with only text 1`] = `
<div
  className="text"
  data-test="Text"
>
  <p
    dangerouslySetInnerHTML={
      Object {
        "__html": "Let's write some tests",
      }
    }
  />
</div>
`;
